reference/conanfile/tools/microsoft.rst

```diff
@@ -310,3 +310,28 @@ Parameters:
 
     * **scope** (Defaulted to ``"build"``): Add the launcher automatically to the ``conanbuild`` launcher. Read more
       in the :ref:`Environment documentation <conan_tools_env_environment_model>`.
+
+
+conan.tools.microsoft.is_msvc()
+-------------------------------
+
+.. code-block:: python
+
+    def is_msvc(conanfile):
+
+Validate ``self.settings.compiler`` for which compiler is being used.
+It returns ``True`` when the host compiler is ``Visual Studio`` or ``msvc``, otherwise, returns ``False``.
+When the ``compiler`` is empty, it raises a ``ConanException``.
```

Comment:
why not just False?

Reply:
Yes, I wouldn't mind if it doesn't raise, and just return False if compiler not defined.

Reply:
It's more like, if you pass ConanFile to is_msvc, but you don't have compiler, it's a violation. When returning False, it's saying, "That's not MSVC", but your compiler is valid.

I understand your point, avoid more exception points, so I'll remove it.